components/core/tests/test-ffi_Value.cpp

```diff
@@ -0,0 +1,195 @@
+ */
+template <typename encoded_variable_t>
+requires(std::is_same_v<encoded_variable_t, eight_byte_encoded_variable_t>
+         || std::is_same_v<encoded_variable_t, four_byte_encoded_variable_t>)
+[[nodiscard]] auto get_encoded_text_ast(std::string_view text
+) -> clp::ir::EncodedTextAst<encoded_variable_t>;
+
+/**
+ * Tests querying the underlying type of the given value against the given type.
+ * @tparam Type The type to query.
+ * @param value The value to test against.
+ */
+template <typename Type>
+auto test_type_probing(Value const& value) -> void;
+
+/**
+ * Tests querying the typed value against the given type.
+ * @tparam Type The type to query.
+ * @param value The value to test against.
+ * @param typed_value The typed value to compare with.
+ */
+template <typename Type>
+auto test_typed_value_probing(Value const& value, Type const& typed_value) -> void;
+
+// Implementation
+
+template <typename encoded_variable_t>
+requires(std::is_same_v<encoded_variable_t, eight_byte_encoded_variable_t>
+         || std::is_same_v<encoded_variable_t, four_byte_encoded_variable_t>)
+auto get_encoded_text_ast(std::string_view text) -> clp::ir::EncodedTextAst<encoded_variable_t> {
+    string logtype;
+    vector<encoded_variable_t> encoded_vars;
+    vector<int32_t> dict_var_bounds;
+    REQUIRE(clp::ffi::encode_message(text, logtype, encoded_vars, dict_var_bounds));
+    REQUIRE(((dict_var_bounds.size() % 2) == 0));
+
+    vector<string> dict_vars;
+    for (size_t i{0}; i < dict_var_bounds.size(); i += 2) {
+        auto const begin_pos{static_cast<size_t>(dict_var_bounds[i])};
+        auto const end_pos{static_cast<size_t>(dict_var_bounds[i + 1])};
+        dict_vars.emplace_back(text.cbegin() + begin_pos, text.cbegin() + end_pos);
+    }
+
+    return clp::ir::EncodedTextAst<encoded_variable_t>{logtype, dict_vars, encoded_vars};
+}
+
+template <typename Type>
+// NOLINTNEXTLINE(readability-function-cognitive-complexity)
+auto test_type_probing(Value const& value) -> void {
+    REQUIRE((std::is_same_v<std::monostate, Type> == value.is_null()));
+    REQUIRE((std::is_same_v<value_int_t, Type> == value.is<value_int_t>()));
+    REQUIRE((std::is_same_v<value_float_t, Type> == value.is<value_float_t>()));
+    REQUIRE((std::is_same_v<value_bool_t, Type> == value.is<value_bool_t>()));
+    REQUIRE((std::is_same_v<string, Type> == value.is<string>()));
+    REQUIRE((std::is_same_v<EightByteEncodedTextAst, Type> == value.is<EightByteEncodedTextAst>()));
+    REQUIRE((std::is_same_v<FourByteEncodedTextAst, Type> == value.is<FourByteEncodedTextAst>()));
+}
+
+template <typename Type>
+// NOLINTNEXTLINE(readability-function-cognitive-complexity)
+auto test_typed_value_probing(Value const& value, Type const& typed_value) -> void {
+    if constexpr (std::is_same_v<value_int_t, Type>) {
+        REQUIRE((value.get_immutable_view<Type>() == typed_value));
+        REQUIRE((std::is_same_v<value_int_t, decltype(value.get_immutable_view<Type>())>));
```

Comment:
```suggestion
        REQUIRE((std::is_same_v<Type, decltype(value.get_immutable_view<Type>())>));
```

This is to slightly reduce the chance of copy-paste errors. Ditto below.